instance/models/instance.py

```diff
@@ -480,9 +513,85 @@ def run_playbook(self):
         return log_lines
 
 
+# Databases ###################################################################
+
+class MySQLInstanceMixin(models.Model):
+    """
+    An instance that uses mysql databases
+    """
+    mysql_user = models.CharField(max_length=16, blank=True)  # 16 chars is mysql maximum
+    mysql_pass = models.CharField(max_length=32, blank=True)
+    mysql_provisioned = models.BooleanField(default=False)
+
+    class Meta:
+        abstract = True
+
+    @property
+    def mysql_database_names(self): #pylint: disable=no-self-use
+        """
+        An iterable of database names
+        """
+        return NotImplementedError
+
+    def provision_mysql(self):
+        """
+        Create mysql user and databases
+        """
+        if settings.INSTANCE_MYSQL_URL_OBJ and not self.mysql_provisioned:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=settings.INSTANCE_MYSQL_URL_OBJ.username,
+                passwd=settings.INSTANCE_MYSQL_URL_OBJ.password or '',
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            cursor = connection.cursor()
+
+            for database in self.mysql_database_names:
+                # We can't use the database name in a parameterized query, the
+                # driver doesn't escape it properly. Se we escape it here instead
+                database_name = connection.escape_string(database).decode()
+                cursor.execute('CREATE DATABASE {0}'.format(database_name))
```

Comment:
Can we make this `'CREATE DATABASE {0} DEFAULT CHARACTER SET utf8'`? It depends on MySQL settings, but by default the character set is not unicode friendly and I had issues when not specifying the default character set on RDS.